### PR TITLE
Added Longest Palindromic Substring [JavaScript]

### LINKED ISSUE #400: Longest Palindromic Substring [JavaScript]
#### This issue is about [Longest Palindromic Substring](http://www.geeksforgeeks.org/longest-palindrome-substring-set-1/).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by Anshuman-Verma**
Fixes #400 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

File: longest_palindromic_substring/longestPalindromicSubstring.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	:return: Longest substring of input string which is a palindrome
+
+	Time complexity: O(n^2)
+	Space complexity: O(1)
+	*/
+	let start = 0;
+	let end = 0;
+
+	for (let i = 0; i < test.length; i++) {
+		let length = Math.max(expandAroundCenter(test, i, i), expandAroundCenter(test, i, i + 1));
```
Comment (Monal5031) [inline longest_palindromic_substring/longestPalindromicSubstring.js]: Please break this line in two for better readability.

File: longest_palindromic_substring/longestPalindromicSubstring.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+			start = i - Math.floor((length - 1) / 2);
+			end = i + Math.floor(length / 2);
+		}
+	}
+	return test.slice(start, end + 1);
+}
+
+function main () {
+	let test = 'geeksforgeeks';
+	console.log('Longest Palindromic Substring of', test, 'is', longestPalindromicSubstring(test));
```
Comment (Monal5031) [inline longest_palindromic_substring/longestPalindromicSubstring.js]: Break this one too.

**Comment by Anshuman-Verma** [comment]
This should be good, @Monal5031 ?

File: longest_palindromic_substring/longestPalindromicSubstring.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	Time complexity: O(n^2)
+	Space complexity: O(1)
+	*/
+	let start = 0;
+	let end = 0;
+
+	for (let i = 0; i < test.length; i++) {
+		let length = Math.max(expandAroundCenter(test, i, i)
+													, expandAroundCenter(test, i, i + 1));
```
Comment (Monal5031) [inline longest_palindromic_substring/longestPalindromicSubstring.js]: This comma would look good in above line :P same about line 43. Also update your branch.

Comment (Anshuman-Verma) [inline longest_palindromic_substring/longestPalindromicSubstring.js]: Alright, makes sense :package:

**Comment by Anshuman-Verma** [comment]
Maybe we've gone too far with short forms @Monal5031 :laughing:

**Comment by Monal5031** [comment]
IMHO short forms are nice to use 😆

**Comment by aviaryan** [comment]
@Anshuman-Verma conflicts!

**Comment by Anshuman-Verma** [comment]
@aviaryan fixed!

**Pull request merged**